code/game/atoms_movable.dm

```diff
@@ -427,7 +427,7 @@
 			continue
 		if(isliving(A))
 			var/mob/living/L = A
-			if(!L.density || L.throwpass)
+			if((!L.density || L.throwpass) && SEND_SIGNAL(A, COMSIG_MOVABLE_LIVING_IMPACT_THROW, src) & COMPONENT_LIVING_THROW_HIT)
```

Comment:
This signals logic needs to be inverted

&& !(SEND_SIGNAL(A, COMSIG_LIVING_PRE_THROW_IMPACT, src) & COMPONENT_PRE_THROW_IMPACT_HIT))